src/code/z_actor.c

```diff
@@ -126,188 +155,910 @@ void func_800B40E0(GlobalContext* globalCtx, Light* light, MtxF* arg2, s32 arg3,
 
     CLOSE_DISPS(globalCtx->state.gfxCtx);
 }
+
+#ifdef NON_MATCHING
+// stack and regalloc
+void ActorShadow_DrawFeet(Actor* actor, Lights* mapper, GlobalContext* globalCtx) {
+    f32 temp_f20_2;
+    f32 temp_f22;
+    f32 temp_f0_2;
+    Vec3f* phi_fp;
+    f32 distToFloor;
+    s32 j;
+    s32 temp_s5;
+
+    MtxF sp13C;
+    MtxF spFC;
+    CollisionPoly* spF8;
+    s32 spF4;
+    f32 floorHeight[2]; // spEC
+
+    s32 temp_a3;
+    Light* sp9C;
+
+    f32 temp_f24;
+
+    s32 phi_s2;
+
+    u8 spD3;
+    s32 spD0;
+    f32* spBC;
+    s32 spB8;
+
+    distToFloor = actor->world.pos.y - actor->floorHeight;
+    if (distToFloor > 0.0f) {
+        f32 shadowScale = actor->shape.shadowScale;
+        u8 shadowAlpha = actor->shape.shadowAlpha;
+        f32 alphaRatio;
+
+        if ((actor->id == ACTOR_PLAYER) && (((Player*)actor)->stateFlags3 & 0x8000)) {
+            f32 temp_f20 = actor->scale.z;
+
+            actor->scale.z += 0.03f * fabsf(Math_CosS(((Player*)actor)->unk_AAA));
+            actor->shape.shadowScale *= 0.2f;
+            alphaRatio = distToFloor * 0.03f;
+            actor->shape.shadowAlpha = actor->shape.shadowAlpha * CLAMP_MIN(alphaRatio, 1.0f);
+            ActorShadow_Draw(actor, mapper, globalCtx, D_04076BC0, NULL);
+            actor->scale.z = temp_f20;
```

Comment:
hang on:
temp_f20 is pulled from actor->scale.z
actor->scale.z is added on line 197
then gets overwritten by re-assigning the temp back to scale.z?

the second line does nothing, because it gets overwritten...?

Reply:
Looks like it's used in `ActorShadow_Draw`.

Reply:
that would make sense, so temp_f20 is storing z so it can be unmodified once passed to ActorShadow_Draw...

Would you call that a `backup` variable or a `savedvalue` or...

Reply:
Yeah. I didn't care much naming variables in NON_MATCHING functions tho :sweat_smile:

Reply:
With this matching, should be able to name temp_f20 now. I normally use something like tmpScaleZ for things like this, but backup or saved as a prefix may be better.